examples/plugins/mockkv/mockkv_backend.cpp

```
@@ -0,0 +1,455 @@
+    nixlMockKVBackendReqH() = default;
+    ~nixlMockKVBackendReqH() = default;
+};
+
+/**
+ * @brief Metadata for MOCKKV memory descriptors
+ *
+ * Stores the mapping between devId and key for later lookups.
+ */
+class nixlMockKVMetadata : public nixlBackendMD {
+public:
+    nixlMockKVMetadata(uint64_t dev_id, std::string key)
```

Cppcheck (2.20.0)

[performance] Function parameter 'key' should be passed by const reference.

(passedByValue)